cocos/2d/components/sprite.ts

```diff
@@ -638,26 +639,24 @@ export class Sprite extends UIRenderer {
     }
 
     private _resized (): void {
-        if (!EDITOR) {
-            return;
-        }
-
-        if (this._spriteFrame) {
-            const actualSize = this.node._uiProps.uiTransformComp!.contentSize;
-            let expectedW = actualSize.width;
-            let expectedH = actualSize.height;
-            if (this._sizeMode === SizeMode.RAW) {
-                const size = this._spriteFrame.originalSize;
-                expectedW = size.width;
-                expectedH = size.height;
-            } else if (this._sizeMode === SizeMode.TRIMMED) {
-                const rect = this._spriteFrame.rect;
-                expectedW = rect.width;
-                expectedH = rect.height;
-            }
+        if (EDITOR) {
+            if (this._spriteFrame) {
```

Comment:
It's no different than the original logic?

Reply:
The refined code is ready; you can check the new commit.